src/test/clojure/xtdb/object_store_test.clj

```diff
@@ -35,9 +39,57 @@
 (defn get-bytes [^ObjectStore obj-store k start len]
   (byte-buf->bytes @(.getObjectRange obj-store k start len)))
 
-(defn in-memory ^Closeable []
-  (->> (ig/prep-key ::os/memory-object-store {})
-       (ig/init-key ::os/memory-object-store)))
+(deftype InMemoryObjectStore [^NavigableMap os]
```

Comment:
Is there much reason to implement this? I'm assuming theres some references to the os-test in memory object store but feels like they should be cleaned up?

Reply:
yep - IIRC main user for this would be the remote buffer pool test, to check that it's actually writing the objects down correctly